### PR TITLE
Enhance/improve build

### PR DESCRIPTION
**Pull request by Hugozys**
This PR replaces Makefile with CMake build system generator to improve the build process.
- Now it will build fruit from source as a shared object and then link that to the calculator, which means we don't need to install pre-built binary anymore
- Document the build steps.

### CODE REVIEW

File: tests/testToken.cc [hunk 1]
```diff
@@ -5,8 +5,8 @@
 #include <string>
 #include <unordered_map>
 
-#include "token.hpp"
-#include "tokengen.hpp"
+#include "srcs/token.hpp"
```
Comment (Hugozys) [inline tests/testToken.cc:8]: This violates coding convention. Will fix them later.

File: tests/testParser.cc [hunk 2]
```diff
@@ -7,9 +7,9 @@
 #include <unordered_map>
 #include <unordered_set>
 
-#include "exception.hpp"
-#include "parser.hpp"
-#include "tokengen.hpp"
+#include "srcs/exception.hpp"
```
Comment (Hugozys) [inline tests/testParser.cc:10]: Same here.

File: main.cpp [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#include <iostream>
+
+int main(int, char**) {
+    std::cout << "Hello, world!\n";
+}
```
Comment (z-rui) [inline main.cpp]: What's this?

Comment (Hugozys) [reply main.cpp]: Oops. Removed.

**Review by z-rui — APPROVED** [review]
LGTM

**Pull request merged**